packages/synapse-react-client/src/components/TableColumnSchemaEditor/TableColumnSchemaEditorUtils.ts

```diff
@@ -396,9 +378,11 @@ export function getTextFieldType(columnType: ColumnTypeEnum) {
   switch (columnType) {
     case ColumnTypeEnum.DOUBLE:
     case ColumnTypeEnum.INTEGER:
-      return 'number'
+      // 'number' is hard to validate, since we allow scientific notation for doubles
+      return 'text'
     case ColumnTypeEnum.DATE:
-      return 'date'
+      // DATE columns actually use datetime precision
+      return 'datetime-local'
     default:
       return 'text'
   }
```

Comment:
Deleting this function in a future PR because it always returns `text` in the context it is used--we use the custom date picker when columnType is `DATE`.